Aleks Shenshin week1 databases homework

shenshin opened the pull request.

jsantiagoh commented on Week1/exercise-2.js:
```diff
@@ -0,0 +1,89 @@
+async function queryWorld() {
+  // show 'limit' number of lines from the result
+  const output = async (query, limit = undefined) => {
+    console.log(await makeQuery(`${query}${limit ? ` LIMIT ${limit}` : ''}`));
```
Nice to have idea: Instead of just logging the entire `RowDataPacket` you could just output the columns that are returned, which would make the output of the program a lot nicer to read.

jsantiagoh commented on Week1/exercise-1.js:
```diff
@@ -0,0 +1,61 @@
+    `CREATE TABLE invitee (
+      invitee_no INT,
+      invitee_name TEXT,
+      invited_by TEXT
```
Fields like names or titles benefit from using `VARCHAR` instead of `TEXT`. There's a few reasons for that but I found them to be summarised correctly [here](https://stackoverflow.com/questions/25300821/difference-between-varchar-and-text-in-mysql)

jsantiagoh commented on Week1/exercise-1.js:
```diff
@@ -0,0 +1,61 @@
+    `CREATE TABLE meeting (
+      meeting_no INT,
+      meeting_title TEXT,
+      starting_time TIME
```
You most likely want to store the date as well as the time on `starting_time`, making `TIME` not the ideal type here.

jsantiagoh reviewed (changes requested):
You did a great job here, there's only a few comments I could find to address and one in general.

The use of `async` always with `await` defeats the purpose of being asynchronous, it's in effect synchronous but adds complexity to read it and maintain it. It's important to get used to write code that is as simple as possible, this helps others (or even yourself) understand it later when it needs to be modified.

shenshin commented:
Hello Santiago! Thanks a lot for your comments! I corrected everything when it was possible. However I could not avoid 'await'-ing in the places where the order of queries does matter

jsantiagoh commented on Week1/exercise-1.js:
```diff
@@ -0,0 +1,115 @@
+  });
+}
+
+async function selectQueries() {
```
As an example of what I mention, you don't have to use an async function and Promises for this, the goal of the exercise is for it to run sequentially in a script, only once. so using `connection.query` makes it simpler

shenshin replied:
Santiago, again I am telling you, I agree that I don't need it when I am creating tables and inserting data. But when I fetching databases and say 'console.log(1)' and then connection.query('...', (err, data) => {console.log(data)}), I first may get displayed the data from the table and then my '1'. May be this is a bad example, but this is what I am always getting when I am trying to deal with async mySql requests. I consider that using async functions is justified here. Can you prove me the opposite?

shenshin replied:
I removed promises completely from this homework.

jsantiagoh reviewed (commented):
Hi Aleks, I don’t mean that you should avoid using await, it’s good if you are using async functions, but my comment was related to that. you don’t need those functions to be asynchronous, because they execute in a script, you can just use connection.query . See https://github.com/mysqljs/mysql#performing-queries

shenshin commented:
Santiago, thank you very much for your feedback! I removed promises where I thought it was possible.

shenshin commented:
I removed promises from everywhere and changed them to callbacks in places where I had to print out the resulting data. Not sure it was the best decision.
